include/swift/AST/Decl.h

```diff
@@ -940,9 +940,27 @@ class alignas(1 << DeclAlignInBits) Decl {
 
   bool isPrivateStdlibDecl(bool treatNonBuiltinProtocolsAsPublic = true) const;
 
-  /// Whether this declaration is weak-imported.
-  bool isWeakImported(ModuleDecl *fromModule,
-                      AvailabilityContext fromContext) const;
+  AvailabilityContext getAvailabilityForLinkage() const;
+
+  /// Whether this declaration or one of it's outer contexts has the
+  /// @_weakLinked attribute.
+  bool isWeakImported() const;
```

Comment:
Can you rename this so it's more obvious what it does? Basically no one should call it directly. (Maybe it should be a helper function in Decl.cpp instead.)

Reply:
Sure, or `isUnconditionallyWeakImported()` maybe.